image-segmentation/python-runnables/download-weights/runnable.py

```diff
@@ -0,0 +1,82 @@
+# This file is the actual code for the Python runnable download-weights
+from dataiku.runnables import Runnable
+import requests
+import dataiku
+import os
+
+try :
+    import pycocotools
+except : 
+    print("##### WARNING ##### Couldn't find pycocotools in installed packages : will proceed to installing the package in plugin codenv")
+    client = dataiku.api_client()
+    codenv = client.get_code_env('python','plugin_image-segmentation_managed')
+    codenv_def = codenv.get_definition()
+    codenv_def['specPackageList'] = 'git+https://github.com/waleedka/coco.git#subdirectory=PythonAPI'
+    codenv.set_definition(codenv_def)
+    
+    print('#'*50)
+    print('Start updating codenv plugin_image-segmentation_managed with git+https://github.com/waleedka/coco.git#subdirectory=PythonAPI')
+    codenv.update_packages()
+    print ('Codenv updated')
+    
+class MyRunnable(Runnable):
+    """The base interface for a Python runnable"""
+
+    def __init__(self, project_key, config, plugin_config):
+        """
+        :param project_key: the project in which the runnable executes
+        :param config: the dict of the configuration of the object
+        :param plugin_config: contains the plugin settings
+        """
+        self.project_key = project_key
+        self.config = config
+        self.plugin_config = plugin_config
+        self.client = dataiku.api_client()
+        
+    def get_progress_target(self):
+        """
+        If the runnable will return some progress info, have this function return a tuple of 
+        (target, unit) where unit is one of: SIZE, FILES, RECORDS, NONE
+        """
+        return None
+
+    def run(self, progress_callback):
+        # Retrieving parameters
+        output_folder_name = self.config.get('outputName', '')
+        
+        # Creating new Managed Folder if needed
+        project = self.client.get_project(self.project_key)
+        output_folder_found = False
+        
+        for folder in project.list_managed_folders():
+            if output_folder_name == folder['name']:
+                output_folder = project.get_managed_folder(folder['id'])
+                output_folder_found = True
+                break
+        
+        if not output_folder_found:
+            output_folder = project.create_managed_folder(output_folder_name)
```

Comment:
If the `output_folder_name` was not specified by user, its value will be an empty string -> error ?

Reply:
put a Try / except raise ValueError